janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -988,6 +992,57 @@ public void testRawQueries() {
         assertCount(numV / strs.length * 2, graph.indexQuery(PINDEX, "text:ducks").properties());
     }
 
+    /**
+     * Tests query parameters with raw indexQuery
+     */
+    @Test
+    public void testRawQueriesWithParameters() throws Exception {
+        if (!supportsLuceneStyleQueries()) return;
+        Parameter asc_sort_p = null;
+        Parameter desc_sort_p = null;
+
+        // ElasticSearch and Solr have different formats for sort parameters
+        if (this.getClass().getSimpleName().toLowerCase().contains("elasticsearch")) {
```

Comment:
Is it possible to use Configuration to get the index backend and do things accordingly?